presentation/src/main/java/com/knocklock/presentation/lockscreen/LockScreenViewModel.kt

```diff
@@ -53,39 +67,217 @@ class LockScreenViewModel @Inject constructor(
+                }
+
+                _recentNotificationList.update {
+                    updatedRecentNotificationList
+                }
+                setRecentNotificationUiFlagMap(removedNotifications.key)
+            }
+        }
+    }
+
+    /**
+     * [_recentNotificationList]에서 전체 제거된 [GroupWithNotification]을 [_recentNotificationUiFlagState]에서도 제거합니다.
+     */
+    private fun removeRecentNotificationUiFlag(groupKey: String) {
+        val removedNotificationUiFlagState = _recentNotificationUiFlagState.value.toMutableMap()
+        removedNotificationUiFlagState.remove(groupKey)
+        _recentNotificationUiFlagState.update { _ ->
+            removedNotificationUiFlagState.toSnapShotStateMap()
+        }
+    }
+
     fun getGroupNotifications(packageManager: PackageManager) {
         viewModelScope.launch {
             notificationRepository.getGroupWithNotificationsWithSorted().collect { groups ->
-                _notificationList.value = NotificationUiState.Success(
+                _oldNotificationList.value = NotificationUiState.Success(
                     groups.map { it.toModel(packageManager) },
                 )
                 launch {
                     groups.forEach { groupNotification ->
                         launch {
                             val key = groupNotification.group.key
                             val flag = groupNotification.notifications.size >= 2
-                            initUiFlagMap(key, flag)
+                            setOldNotificationUiFlagMap(key, flag)
                         }
                     }
                 }
             }
         }
     }
 
+    fun saveRecentNotificationToDatabase() {
+        viewModelScope.launch {
+            val stack = Stack<GroupWithNotification>()
+            _recentNotificationList.value.forEach {
+                stack.add(it)
+            }
+            while (stack.isNotEmpty()) {
+                val groupWithNotificationStack = stack.pop()
+                _recentNotificationList.update {
+                    stack.toList()
+                }
+                notificationRepository.insertGroup(
+                    groupWithNotificationStack.toModel().group,
+                )
+                notificationRepository.insertNotifications(
+                    *groupWithNotificationStack.notifications.map { notification ->
+                        notification.toModel()
+                    }.toTypedArray(),
+                )
+            }
+        }
+    }
+
     fun setComposeScreenState(composeScreenState: ComposeScreenState) {
         _composeScreenState.value = composeScreenState
     }
 
-    private fun initUiFlagMap(key: String, flag: Boolean) = with(_notificationUiFlagStateState.value) {
+    fun addRecentNotification(notification: NotificationModel, packageManager: PackageManager) {
+        val isExisted = _recentNotificationList.value.count {
+            it.group.key == notification.groupKey
+        } == 1
```

Comment:
```suggestion
        val isExisted = _recentNotificationList.value.any{
            it.group.key == notification.groupKey
        } 
```

isExisted 용도라면 any를 사용해도 좋을 것 같네요~!

Reply:
@jaeryo2357 끝나고 반영할께영

Reply:
@jaeryo2357 [a01d37b](https://github.com/Knock-Lock/Knock-Lock/pull/136/commits/a01d37ba8977ac60ecf61cbbf7484cb1a3a34fc4) 반영~!